### PR TITLE
more login capabilities

### PR DESCRIPTION
**Pull request by stefanfoulis**
- [x] FE: styling and html structure @Brankub 
- [x] FE: all styling files as part of the package (no more using static.aldryn.com!) 
- [x] BE: remove ``CMSCLOUD_STATIC_URL`` @stefanfoulis
- [x] BE: make sure ``next=`` works as expected @stefanfoulis
- [x] BE: handle case where user is already logged in and ends up on the login view @stefanfoulis
- [x] test it with all the combinations (live/stage/local and the different settings) @stefanfoulis
- [x] auto-login if only SSO is configured
- [ ] test with Django 1.6 and 1.9. Python 3

### CODE REVIEW

File: aldryn_sso/templates/aldryn_sso/includes/local_development_login_form.html [hunk 1]
```diff
@@ -33,8 +31,8 @@ <h3 class="page-status">Login</h3>
...
             {% endwith %}
             <input type="hidden" name="next" value="{{ next }}" />
-            <input class="btn btn-marketing" type="submit" value="Sign in">
+            <input class="btn btn-marketing" type="submit" value="{% trans 'Sign in' %}">
+            <div class="spacer"></div>
         </form>
-        <br />
     </div>
-{% endblock content %}
+</div>
```
Comment (Chive) [inline aldryn_sso/templates/aldryn_sso/includes/local_development_login_form.html]: really? every PR lately 🐙

File: aldryn_sso/templates/aldryn_sso/includes/sso_login_button.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{% load i18n admin_static %}
+{% url 'simple-sso-login' as sso_login_url %}
+<a
+    class="btn btn-marketing with-brand-icon btn-block"
+    href="{{ sso_login_url }}{% if request.GET.next %}?next={{ request.GET.next|urlencode }}{% endif %}">
+    <span aria-hidden="true" class="aldryn-icon"></span>
+    {% trans "Sign in with Aldryn" %}
+</a>
```
Comment (Chive) [inline aldryn_sso/templates/aldryn_sso/includes/sso_login_button.html]: 🐙

File: aldryn_sso/views.py [hunk 3]
```diff
@@ -1,27 +1,51 @@
...
+from django.views.decorators.csrf import csrf_protect
+from django.views.decorators.debug import sensitive_post_parameters
 from django.views.generic import CreateView
 
 from .forms import CreateUserForm, LoginAsForm
 
 
+def get_shared_context():
+    context = {}
+    if settings.ALDRYN_SSO_ENABLE:
```
Comment (Chive) [inline aldryn_sso/views.py]: ``enableD``?

File: aldryn_sso/views.py [hunk 4]
```diff
@@ -1,27 +1,51 @@
...
+        context.update({
+            'aldryn_sso_enable': True,
+        })
+    if settings.ALDRYN_SSO_ENABLE_STANDARD_LOGIN:
```
Comment (Chive) [inline aldryn_sso/views.py]: same ^

File: aldryn_sso/views.py [hunk 5]
```diff
@@ -1,27 +1,51 @@
...
+        context.update({
+            'aldryn_sso_standard_login_form': AuthenticationForm(),
+            'aldryn_sso_enable_standard_login': True,
+        })
+    if settings.ALDRYN_LOCALDEV_ENABLE:
```
Comment (Chive) [inline aldryn_sso/views.py]: same ^

File: aldryn_sso/views.py [hunk 6]
```diff
@@ -1,27 +1,51 @@
...
+        django.contrib.auth.REDIRECT_FIELD_NAME,
+        request.GET.get(django.contrib.auth.REDIRECT_FIELD_NAME, '')
     )
     return redirect_to
 
 
 def get_redirect_url(request, fallback=None):
-    # don't use request.REQUEST
-    # is deprecated
+    # don't use request.REQUEST because it is deprecated.
```
Comment (Chive) [inline aldryn_sso/views.py]: why even keep this?

File: aldryn_sso/views.py [hunk 7]
```diff
@@ -66,13 +90,45 @@ class CreateUserView(CreateView):
...
+        )
+    ):
+        # The aldryn SSO button would be the only thing on the page. So we just
+        # initiate the login without further ado.
+        sso_url = '{}?{}'.format(
+            reverse('simple-sso-login'),
+            urlencode(dict(next=next_url)),
+        )
+        return HttpResponseRedirect(sso_url)
+    return django.contrib.auth.views.login(request, **kwargs)
```
Comment (Chive) [inline aldryn_sso/views.py]: steeeefan 🐙

File: aldryn_sso/views.py (hunk 3 above)
Comment (stefanfoulis) [inline aldryn_sso/views.py]: or maybe better add something after it, so it matches the other settings:
- ``ALDRYN_SSO_ENABLE`` -> ``ALDRYN_SSO_ENABLE_LOGIN``
- ``ALDRYN_SSO_ENABLE_STANDARD_LOGIN``: ok
- ``ALDRYN_LOCALDEV_ENABLE`` -> ???

**Pull request merged**